# Classification Student Performance Dataset

Project ini bertujuan untuk memprediksi kelulusan siswa berdasarkan nilai akademik menggunakan metode classification dengan Logistic Regression.

In [13]:
from google.colab import files
uploaded = files.upload()

Saving student-mat.csv to student-mat (1).csv


In [14]:
import pandas as pd

df = pd.read_csv("student-mat.csv", sep=",")
df.head()

school sex  age address famsize Pstatus  Medu  Fedu     Mjob      Fjob  ...  \
0     GP   F   18       U     GT3       A     4     4  at_home   teacher  ...   
1     GP   F   17       U     GT3       T     1     1  at_home     other  ...   
2     GP   F   15       U     LE3       T     1     1  at_home     other  ...   
3     GP   F   15       U     GT3       T     4     2   health  services  ...   
4     GP   F   16       U     GT3       T     3     3    other     other  ...   

  famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0      4        3      4     1     1      3        6   5   6   6  
1      5        3      3     1     1      3        4   5   5   6  
2      4        3      2     2     3      3       10   7   8  10  
3      3        2      2     1     1      5        2  15  14  15  
4      4        3      2     1     2      5        4   6  10  10  

[5 rows x 33 columns]

## Deskripsi Dataset

Dataset Student Performance berisi data akademik siswa seperti jenis kelamin, usia, waktu belajar, dan nilai ujian. Dataset terdiri dari 395 data siswa dan digunakan untuk memprediksi kelulusan siswa.

In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 395 entries, 0 to 394
Data columns (total 33 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   school      395 non-null    object
 1   sex         395 non-null    object
 2   age         395 non-null    int64 
 3   address     395 non-null    object
 4   famsize     395 non-null    object
 5   Pstatus     395 non-null    object
 6   Medu        395 non-null    int64 
 7   Fedu        395 non-null    int64 
 8   Mjob        395 non-null    object
 9   Fjob        395 non-null    object
 10  reason      395 non-null    object
 11  guardian    395 non-null    object
 12  traveltime  395 non-null    int64 
 13  studytime   395 non-null    int64 
 14  failures    395 non-null    int64 
 15  schoolsup   395 non-null    object
 16  famsup      395 non-null    object
 17  paid        395 non-null    object
 18  activities  395 non-null    object
 19  nursery     395 non-null    object
 20  higher    

In [16]:
df["lulus"] = (df["G3"] >= 10).astype(int)
df[["G3","lulus"]].head()

G3  lulus
0   6      0
1   6      0
2  10      1
3  15      1
4  10      1

## Pembuatan Target Classification

Variabel target "lulus" dibuat berdasarkan nilai akhir (G3). Jika nilai G3 ≥ 10 maka siswa dinyatakan lulus, sedangkan nilai di bawah 10 dinyatakan tidak lulus.

In [17]:
X = df[["G1","G2"]]
y = df["lulus"]

In [18]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42
)

In [19]:
from sklearn.linear_model import LogisticRegression

model = LogisticRegression()
model.fit(X_train, y_train)

LogisticRegression()

## Model Classification

Model Logistic Regression digunakan untuk memprediksi kelulusan siswa berdasarkan nilai sebelumnya yaitu G1 dan G2.

In [20]:
from sklearn.metrics import accuracy_score

pred = model.predict(X_test)
accuracy_score(y_test, pred)

0.9113924050632911

## Hasil dan Evaluasi Model

Hasil pengujian menunjukkan akurasi sekitar 91%. Hal ini menunjukkan bahwa model mampu memprediksi kelulusan siswa dengan tingkat ketepatan yang tinggi.